### ISSUE TITLE
Use Cgo to support C modules

### ISSUE THREAD

**Issue by AverseABFun** [go-python/gpython#235]
I'll be honest; I'm completely new to this project and know next to nothing about it, so if there's a known reason Cgo wouldn't work, sorry. However, I think that using Cgo could be an easy(ish) way to support other modules, however likely with some changes for how Gpython works over others.

**Comment by ncw** [comment]
We could certainly initialise and run python modules written in C, however we'd have to re-implement all the C services that they rely on so that the Go equivalent got called. Given that the implementation of Python objects is completely different to that of CPython I'd put this in the very hard but not quite impossible category!

**Comment by sbinet** [comment]
another option (possibly as difficult or work intensive) would be to emulate the C API as PyPy did with its cpyext module:

- https://pypy.org/posts/2018/09/inside-cpyext-why-emulating-cpython-c-8083064623681286567.html
- https://pythoncapi.readthedocs.io/index.html
- https://pythoncapi.readthedocs.io/cpyext.html

**Comment by corona10** [comment]
My personal suggestion is targeting HPy instead of C API itself, C API expose too much of implementation detail from CPython.
https://hpyproject.org/

**Comment by corona10** [comment]
FYI, GraalPython and PyPy success to support HPy

**Comment by sbinet** [comment]
yes, I was about to suggest HPy as well (but was refreshing my memory a bit about it before posting :})

anyways: whatever the avenue gpython chooses, it's a very long and hard road, which (to be honest) is probably not sustainable with the current workforce available for this project (sadly, but it is what it is)
